In [1]:
import numpy as np

Calculate the mean, median, mode and standard deviation for the problem
statements 1& 2.
### Problem Statement 1:
The marks awarded for an assignment set for a Year 8 class of 20 students were as
follows:
6 7 5 7 7 8 7 6 9 7 4 10 6 8 8 9 5 6 4 8

In [2]:
l= [6,7, 5, 7, 7, 8, 7, 6, 9, 7, 4, 10, 6, 8, 8, 9, 5, 6, 4, 8]
l = np.array(l)
print("mean = ",np.mean(l))
print("median = ",np.median(l))
print("std = ",np.std(l))


mean =  6.85
median =  7.0
std =  1.5898113095584647


#### Problem Statement 2:
The number of calls from motorists per day for roadside service was recorded for a
particular month:
28, 122, 217, 130, 120, 86, 80, 90, 140, 120, 70, 40, 145, 113, 90, 68, 174, 194, 170,
100, 75, 104, 97, 75,
123, 100, 75, 104, 97, 75, 123, 100, 89, 120, 109

In [3]:
l= np.array([28, 122, 217, 130, 120, 86, 80, 90, 140, 120, 70, 40, 145, 113, 90, 68, 174, 194, 170, 100, 75, 104, 97, 75, 123, 100, 75, 104, 97, 75, 123, 100, 89, 120, 109])
print("mean = ",np.mean(l))
print("median = ",np.median(l))
print("std = ",np.std(l))



mean =  107.51428571428572
median =  100.0
std =  38.77287080168403


##### Problem Statement 3:
The number of times I go to the gym in weekdays, are given below along with its
associated probability:
x = 0, 1, 2, 3, 4, 5
f(x) = 0.09, 0.15, 0.40, 0.25, 0.10, 0.01
Calculate the mean no. of workouts in a week. Also evaluate the variance involved in
it.

In [10]:
from functools import reduce


In [14]:
x = (0, 1, 2, 3, 4, 5)
p = (0.09, 0.15, 0.40, 0.25, 0.10, 0.01)
px=tuple(map(lambda x,y:x*y, x,p))
px
mean = reduce(lambda a,b:a+b,px)
print("mean = ",mean)
std = np.std(px)
variance = np.sqrt(std)
print("variance=",variance)

mean =  2.15
variance= 0.5663020602248394


#### Problem Statement 4:
Let the continuous random variable D denote the diameter of the hole drilled in an
aluminum sheet. The target diameter to be achieved is 12.5mm. Random
disturbances in the process often result in inaccuracy.
Historical data shows that the distribution of D can be modelled by the PDF (d) =
20e−20(d−12.5), d ≥ 12.5. If a part with diameter > 12.6 mm needs to be scrapped,
what is the proportion of those parts? What is the CDF when the diameter is of 11
mm? What is your conclusion regarding the proportion of scraps?

### Problem Statement 5:

A company manufactures LED bulbs with a faulty rate of 30%. If I randomly select 6
chosen LEDs, what is the probability of having 2 faulty LEDs in my sample?
Calculate the average value of this process. Also evaluate the standard deviation
associated with it.

given
fault rate= 30% means if manufacturer is manudfacturing the 100 LED then 30 bulb marked defected as defected
number of sample(n) = 6
x=2
p(x) = ncx * p^x * (1-p)^(n-x)
p(x) = 6c2 * 0.3^2 * 0.7^4

In [15]:
0.3 **2

0.09

In [18]:
p_2=15 * (0.3 ** 2) * (0.7 ** 4)
p_2

0.3241349999999999

so here 32.4% chances are there that 2 bulb will be faulty out of 6 bulb

In [22]:
sd =(2-0.3) **2 * p_2
#standard daviation = sd
sd

0.9367501499999996

#### Problem Statement 6:

Gaurav and Barakha are both preparing for entrance exams. Gaurav attempts to
solve 8 questions per day with a correction rate of 75%, while Barakha averages
around 12 questions per day with a correction rate of 45%. What is the probability
that each of them will solve 5 questions correctly? What happens in cases of 4 and 6
correct solutions? What do you infer from it? What are the two main governing
factors affecting their ability to solve questions correctly? Give a pictorial
representation of the same to validate your answer.

probablity to solve the correct solution by gaurav:- 75%=0.75
probablity to solve the correct solution by Barakha:- 45%=0.45
n_gaurav = 8
n_barkha =12
likehood_events(x) = 5
now lets calculate the % of solving the 5 que correctly by each of them
p(x) = ncx * p^x * (1-p)^(n-x)

In [27]:
#p_gaurav = 8c5 * 0.75 **5 *0.25 **3
p_gaurav = 56 * 0.75 **5 *0.25 **3
p_barkha = 792 * 0.45 **5 * 0.55 ** 7

print("probablity to solve the 5 que by gaurav=",p_gaurav)
print("probablity to solve the 5 que by barkha=",p_barkha)


probablity to solve the 5 que by gaurav= 0.2076416015625
probablity to solve the 5 que by barkha= 0.22249823843265837


In [28]:
#What happens in cases of 4 and 6 correct solutions?
#p_gaurav = 8c4 * 0.75 **4 *0.25 **4
#p_gaurav = 12c6 * 0.75 **6 *0.25 **6
p_gaurav = 70 * 0.75 **4 *0.25 **4
p_barkha = 924 * 0.45 **6 * 0.55 ** 6

print("probablity to solve the 4 que by gaurav=",p_gaurav)
print("probablity to solve the 6 que by barkha=",p_barkha)

probablity to solve the 4 que by gaurav= 0.086517333984375
probablity to solve the 6 que by barkha= 0.2123846821402648


In [ ]:
#with the above calculation I get to know after adding the both the observation it almost = 1

##### Problem Statement 7:

Customers arrive at a rate of 72 per hour to my shop. What is the probability of k
customers arriving in 4 minutes? a) 5 customers, b) not more than 3 customers, c)
more than 3 customers. Give a pictorial representation of the same to validate your
answer.

with the above problem statement given data
in every 60 min 72 customers are visiting in his shop
so in 4  min (72/60)*4~=5 customers

In [29]:
# a) 5 customers in 4 min
p()

4.8

##### Problem Statement 8:

I work as a data analyst in Aeon Learning Pvt. Ltd. After analyzing data, I make
reports, where I have the efficiency of entering 77 words per minute with 6 errors per
hour. What is the probability that I will commit 2 errors in a 455-word financial report?
What happens when the no. of words increases/decreases (in case of 1000 words,
255 words)?
How is the λ affected?
How does it influence the PMF?
Give a pictorial representation of the same to validate your answer.

#### Problem Statement 9:

Let the continuous random variable D denote the diameter of the hole drilled in an
aluminum sheet. The target diameter to be achieved is 12.5mm. Random
disturbances in the process often result in inaccuracy.
Historical data shows that the distribution of D can be modelled by the PDF, f(d) =
20e−20(d−12.5), d ≥ 12.5. If a part with diameter > 12.6 mm needs to be scrapped,
what is the proportion of those parts? What is the CDF when the diameter is of 11
mm?
What is the conclusion of this experiment?

### Problem Statement 11:

The current flow in a copper wire follow a normal distribution with a mean of 10 mA
and a variance of 4 (mA)2.
What is the probability that a current measurement will exceed 13 mA? What is the
probability that a current measurement is between 9 and 11mA? Determine the
current measurement which has a probability of 0.98.

mu = 10ma
sigma = 4
x = 13 mA
z= (x-mu)/sigma
p(z)= ?


In [32]:
z = (13-10)/4
z

0.75

In [33]:
# by the Z distribution table 
#p(Z)= 77%

#the probability that a current measurement will exceed 13 mA is 77%

30.85-15.87

14.980000000000002

<img src="problem_11_1.jpeg" height=50 alt-text="prob_11_1 logo">

<img src="pro_11_2.jpeg" height=50 alt-text="prob_11_1 logo">

##### Problem Statement 12:

The shaft in a piston has its diameter normally distributed with a mean of 0.2508 inch
and a standard deviation of 0.0005 inch. The specifications of the shaft are 0.2500 ∓
0.0015 inch. What proportion of shafts are in sync with the specifications? If the
process is centered so that the mean is equal to the target value of 0.2500, what
proportion of shafts conform to the new specifications? What is your conclusion from
this experiment?

In [ ]:
np.exp